pkg/patterns/addon/pkg/status/accessors.go

```diff
@@ -0,0 +1,88 @@
+package status
+
+import (
+	"fmt"
+	"reflect"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func GetConditions(instance runtime.Object) ([]metav1.Condition, error) {
+	statusVal := reflect.ValueOf(instance).Elem().FieldByName("Status")
+	if !statusVal.IsValid() {
+		return nil, fmt.Errorf("status field not found")
+	}
+	conditionsVal := statusVal.FieldByName("Conditions")
+	if !conditionsVal.IsValid() {
+		return nil, fmt.Errorf("status.conditions field not found")
+		//  &MissingConditionsErr{Object: instance}
+	}
+
+	v := conditionsVal.Interface()
+	conditions, ok := v.([]metav1.Condition)
+	if !ok {
+		return nil, fmt.Errorf("unexpecetd type for status.conditions; got %T, want []metav1.Condition", v)
+	}
+	return conditions, nil
+
+	// if v, ok := instance.(addonsv1alpha1.ConditionGetterSetter); ok {
+	// 	return v.GetConditions(), nil
+	// }
+	// instance cannot type assert to *unstructured.Unstructured directly.
+	// v, err := runtime.DefaultUnstructuredConverter.ToUnstructured(instance)
+	// if err != nil {
+	// 	return nil, err
+	// }
+	// unstructConditions, _, err := unstructured.NestedSlice(v, "status", "conditions")
+	// if err != nil {
+	// 	return nil, fmt.Errorf("unable to get status.conditions from unstuctured: %v", err)
+	// }
+	// var conditions []metav1.Condition
+	// for _, ucond := range unstructConditions {
+	// 	var addonCond metav1.Condition
+	// 	err = runtime.DefaultUnstructuredConverter.FromUnstructured(ucond.(map[string]interface{}), &addonCond)
+	// 	if err != nil {
+	// 		return nil, err
+	// 	}
+	// 	conditions = append(conditions, addonCond)
+	// }
+	// return conditions, nil
+}
+
+// type MissingConditionsErr struct {
+// 	Object runtime.Object
+// }
+
+// func (e *MissingConditionsErr) Error() string {
+// 	return fmt.Sprintf("unable to find `status.condition` in %T", e.Object)
+// }
+
+func SetConditions(instance runtime.Object, newConditions []metav1.Condition) error {
+	// if v, ok := instance.(addonsv1alpha1.ConditionGetterSetter); ok {
+	// 	v.SetConditions(conditions)
+	// 	return nil
+	// }
+	// newConditionsVal := reflect.ValueOf(conditions).FieldByName("Conditions")
+	statusVal := reflect.ValueOf(instance).Elem().FieldByName("Status")
+	if !statusVal.IsValid() {
+		// Status not ready.
+		return fmt.Errorf("status field not found")
+	}
+	conditionsVal := statusVal.FieldByName("Conditions")
+	if !conditionsVal.IsValid() {
+		return fmt.Errorf("status.conditions field not found")
+		//  &MissingConditionsErr{Object: instance}
+	}
```

Comment:
We don't want to return a generic error because otherwise the caller method will propagate the error rather than updating the object `status`. The result is that the object `status.condition` will never be updated  
 https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/pull/319/files#diff-eb527f90ab4b8fa744a6601d127c711bdde2d3cd9956532a736738203a4fc94fR105

Similar to why we do not return error  if status does not exist yet (`if !statusVal.IsValid()` ).